// @vonage/messages/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Messages = exports.CustomMessage = exports.TemplateMessage = exports.ViberText = exports.MessengerText = exports.WhatsAppText = exports.MessengerFile = exports.WhatsAppFile = exports.MessengerVideo = exports.WhatsAppVideo = exports.MMSVideo = exports.WhatsAppAudio = exports.MessengerAudio = exports.MMSAudio = exports.MMSVcard = exports.ViberImage = exports.MessengerImage = exports.WhatsAppImage = exports.MMSImage = exports.SMS = void 0;
// Concrete classes
var SMS_1 = require("./classes/SMS/SMS");
Object.defineProperty(exports, "SMS", { enumerable: true, get: function () { return SMS_1.SMS; } });
var Image_1 = require("./classes/MMS/Image");
Object.defineProperty(exports, "MMSImage", { enumerable: true, get: function () { return Image_1.Image; } });
var Image_2 = require("./classes/WhatsApp/Image");
Object.defineProperty(exports, "WhatsAppImage", { enumerable: true, get: function () { return Image_2.Image; } });
var Image_3 = require("./classes/Messenger/Image");
Object.defineProperty(exports, "MessengerImage", { enumerable: true, get: function () { return Image_3.Image; } });
var Image_4 = require("./classes/Viber/Image");
Object.defineProperty(exports, "ViberImage", { enumerable: true, get: function () { return Image_4.Image; } });
var Vcard_1 = require("./classes/MMS/Vcard");
Object.defineProperty(exports, "MMSVcard", { enumerable: true, get: function () { return Vcard_1.Vcard; } });
var Audio_1 = require("./classes/MMS/Audio");
Object.defineProperty(exports, "MMSAudio", { enumerable: true, get: function () { return Audio_1.Audio; } });
var Audio_2 = require("./classes/Messenger/Audio");
Object.defineProperty(exports, "MessengerAudio", { enumerable: true, get: function () { return Audio_2.Audio; } });
var Audio_3 = require("./classes/WhatsApp/Audio");
Object.defineProperty(exports, "WhatsAppAudio", { enumerable: true, get: function () { return Audio_3.Audio; } });
var Video_1 = require("./classes/MMS/Video");
Object.defineProperty(exports, "MMSVideo", { enumerable: true, get: function () { return Video_1.Video; } });
var Video_2 = require("./classes/WhatsApp/Video");
Object.defineProperty(exports, "WhatsAppVideo", { enumerable: true, get: function () { return Video_2.Video; } });
var Video_3 = require("./classes/Messenger/Video");
Object.defineProperty(exports, "MessengerVideo", { enumerable: true, get: function () { return Video_3.Video; } });
var File_1 = require("./classes/WhatsApp/File");
Object.defineProperty(exports, "WhatsAppFile", { enumerable: true, get: function () { return File_1.File; } });
var File_2 = require("./classes/Messenger/File");
Object.defineProperty(exports, "MessengerFile", { enumerable: true, get: function () { return File_2.File; } });
var Text_1 = require("./classes/WhatsApp/Text");
Object.defineProperty(exports, "WhatsAppText", { enumerable: true, get: function () { return Text_1.Text; } });
var Text_2 = require("./classes/Messenger/Text");
Object.defineProperty(exports, "MessengerText", { enumerable: true, get: function () { return Text_2.Text; } });
var Text_3 = require("./classes/Viber/Text");
Object.defineProperty(exports, "ViberText", { enumerable: true, get: function () { return Text_3.Text; } });
var TemplateMessage_1 = require("./classes/WhatsApp/TemplateMessage");
Object.defineProperty(exports, "TemplateMessage", { enumerable: true, get: function () { return TemplateMessage_1.TemplateMessage; } });
var CustomMessage_1 = require("./classes/WhatsApp/CustomMessage");
Object.defineProperty(exports, "CustomMessage", { enumerable: true, get: function () { return CustomMessage_1.CustomMessage; } });
// Main client
var messages_1 = require("./messages");
Object.defineProperty(exports, "Messages", { enumerable: true, get: function () { return messages_1.Messages; } });
