// @vonage/voice/dist/classes/NCCO/Input.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Input = void 0;
class Input {
    constructor(dtmf, speech, eventUrl, eventMethod) {
        this.action = 'input';
        this.type = [];
        this.eventUrl = [];
        if (dtmf) {
            this.type.push('dtmf');
            this.dtmf = dtmf;
        }
        if (speech) {
            this.type.push('speech');
            this.speech = speech;
        }
        if (eventUrl) {
            this.eventUrl.push(eventUrl);
        }
        if (eventMethod) {
            this.eventMethod = eventMethod;
        }
        if (this.type.length === 0) {
            throw new TypeError('Input action must have at least either DTMF or Speech settings');
        }
    }
    serializeToNCCO() {
        const data = {
            action: this.action,
            type: this.type,
        };
        if (this.dtmf) {
            data.dtmf = this.dtmf;
        }
        if (this.speech) {
            data.speech = this.speech;
        }
        if (this.eventUrl) {
            data.eventUrl = this.eventUrl;
        }
        if (this.eventMethod) {
            data.eventMethod = this.eventMethod;
        }
        return data;
    }
}
exports.Input = Input;
