// @vonage/voice/dist/classes/NCCO/Talk.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Talk = void 0;
class Talk {
    constructor(text, bargeIn, loop, level, language, style, premium) {
        this.action = 'talk';
        this.text = text;
        if (bargeIn) {
            this.bargeIn = bargeIn;
        }
        if (loop) {
            this.loop = loop;
        }
        if (level) {
            this.level = level;
        }
        if (language) {
            this.language = language;
        }
        if (style) {
            this.style = style;
        }
        if (premium) {
            this.premium = premium;
        }
    }
    serializeToNCCO() {
        const data = {
            action: this.action,
            text: this.text,
        };
        if (this.loop) {
            data.loop = this.loop;
        }
        if (this.bargeIn) {
            data.bargeIn = this.bargeIn;
        }
        if (this.level) {
            data.level = this.level;
        }
        if (this.language) {
            data.language = this.language;
        }
        if (this.style) {
            data.style = this.style;
        }
        if (this.premium) {
            data.premium = this.premium;
        }
        return data;
    }
}
exports.Talk = Talk;
