// @vonage/applications/dist/applications.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Applications = void 0;
const server_client_1 = require("@vonage/server-client");
class Applications extends server_client_1.Client {
    constructor() {
        super(...arguments);
        this.authType = server_client_1.AuthenticationType.BASIC;
    }
    createApplication(application) {
        return __awaiter(this, void 0, void 0, function* () {
            const resp = yield this.sendPostRequest(`${this.config.apiHost}/v2/applications`, application);
            return resp.data;
        });
    }
    deleteApplication(applicationId) {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.sendDeleteRequest(`${this.config.apiHost}/v2/applications/${applicationId}`);
        });
    }
    getApplication(applicationId) {
        return __awaiter(this, void 0, void 0, function* () {
            const resp = yield this.sendGetRequest(`${this.config.apiHost}/v2/applications/${applicationId}`);
            return resp.data;
        });
    }
    listApplications(filter) {
        return __awaiter(this, void 0, void 0, function* () {
            const resp = yield this.sendGetRequest(`${this.config.apiHost}/v2/applications`, filter);
            return resp.data;
        });
    }
    updateApplication(application) {
        return __awaiter(this, void 0, void 0, function* () {
            const resp = yield this.sendPutRequest(`${this.config.apiHost}/v2/applications/${application.id}`, application);
            return resp.data;
        });
    }
}
exports.Applications = Applications;
