// @vonage/voice/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Talk = exports.NCCOBuilder = exports.Input = exports.OutboundCallWithAnswerURL = exports.OutboundCallWithNCCO = exports.Voice = void 0;
var voice_1 = require("./voice");
Object.defineProperty(exports, "Voice", { enumerable: true, get: function () { return voice_1.Voice; } });
var OutboundCallWithNCCO_1 = require("./classes/OutboundCallWithNCCO");
Object.defineProperty(exports, "OutboundCallWithNCCO", { enumerable: true, get: function () { return OutboundCallWithNCCO_1.OutboundCallWithNCCO; } });
var OutboundCallWithAnswerURL_1 = require("./classes/OutboundCallWithAnswerURL");
Object.defineProperty(exports, "OutboundCallWithAnswerURL", { enumerable: true, get: function () { return OutboundCallWithAnswerURL_1.OutboundCallWithAnswerURL; } });
var ncco_1 = require("./ncco");
Object.defineProperty(exports, "Input", { enumerable: true, get: function () { return ncco_1.Input; } });
Object.defineProperty(exports, "NCCOBuilder", { enumerable: true, get: function () { return ncco_1.NCCOBuilder; } });
Object.defineProperty(exports, "Talk", { enumerable: true, get: function () { return ncco_1.Talk; } });
